mettle/src/process.c

```diff
@@ -365,16 +357,16 @@ static struct process * process_create(struct procmgr *mgr,
 		return NULL;
 	}
 
-	pid_t pid = fork();
+	int master;
+
+	pid_t pid = forkpty(&master, NULL, NULL, NULL);
```

Comment:
Do we want to honor the semantics of channelized here? i.e. Not opening _all_ channels in ptys?

https://github.com/rapid7/metasploit-payloads/blob/45e98c85a3dc2b55d8e907a87c0555a89e3a1aa3/python/meterpreter/ext_server_stdapi.py#L1156

Reply:
Looks like channelized isn't related to pty in any capacity, but that's what the Python meterpreter assumes:
https://github.com/rapid7/metasploit-payloads/blob/master/python/meterpreter/ext_server_stdapi.py#L1156

Caught up with Dean, and. I think we might need a separate flag for requesting a PTY